recomendator/recomendations/models.py

```diff
@@ -0,0 +1,95 @@
+import uuid
+
+from django.db import models
+from django.contrib.auth.models import UserManager, AbstractUser
+
+
+
+# Create your models here.
+class LibraryUser(AbstractUser):
+    objects = UserManager()
+
+
+class Genre(models.Model):
+    """Model representing a book genre (e.g. Science Fiction, Non Fiction)."""
+
+    name = models.CharField(
+        max_length=200,
+        help_text="Enter a book genre (e.g. Science Fiction, French Poetry etc.)"
+        )
+
+    def __str__(self):
+        """String for representing the Model object (in Admin site etc.)"""
+        return self.name
+
+
+class Book(models.Model):
+    """Model representing a book (but not a specific copy of a book)."""
+    CQRS_ID = 'book'
+
+    title = models.CharField(max_length=200)
+    author = models.ForeignKey('Author', on_delete=models.SET_NULL, null=True)
+    # Foreign Key used because book can only have one author, but authors can have multiple books
+    # Author as a string rather than object because it hasn't been declared yet in file.
+    summary = models.TextField(max_length=1000, help_text="Enter a brief description of the book")
+    isbn = models.CharField('ISBN', max_length=13,
+                            unique=True,
+                            help_text='13 Character <a href="https://www.isbn-international.org/content/what-isbn'
+                                      '">ISBN number</a>')
+    genre = models.ManyToManyField(Genre, help_text="Select a genre for this book")
+    # ManyToManyField used because a genre can contain many books and a Book can cover many genres.
+    # Genre class has already been defined so we can specify the object above.
+    language = models.IntegerField()
+    
+    class Meta:
+        ordering = ['title', 'author']
+
+    def __str__(self):
+        """String for representing the Model object."""
+        return self.title
+
+class BookInstance(models.Model):
+    """Model representing a specific copy of a book (i.e. that can be borrowed from the library)."""
+    CQRS_SERIALIZER = 'recomendator.serializers.BookInstanceSerializer'
+
+    id = models.UUIDField(primary_key=True, default=uuid.uuid4,
+                          help_text="Unique ID for this particular book across whole library")
+    book = models.ForeignKey('Book', on_delete=models.RESTRICT, null=True)
+    imprint = models.CharField(max_length=200)
+    due_back = models.DateField(null=True, blank=True)
+    borrower = models.ForeignKey(LibraryUser, on_delete=models.SET_NULL, null=True, blank=True)
+
+    @property
+    def pk(self):
```

Comment:
Isn't django doing this automatically? At least in the catalog it's working

Reply:
If you mean the pk property, it does, but CQRS will fail by this as the id is an uuid and it does'nt know how to serialize it and neither can be fixed in the custom serializer